tiledb/sm/c_api/tiledb.h

```diff
@@ -3457,6 +3457,43 @@ TILEDB_EXPORT int32_t tiledb_array_schema_has_attribute(
 TILEDB_EXPORT int32_t tiledb_array_schema_dump(
     tiledb_ctx_t* ctx, const tiledb_array_schema_t* array_schema, FILE* out);
 
+/**
+ * Returns the array schema in ASCII format as a string.
+ *
+ * **Example:**
+ *
+ * The following retrieves the array-schema dump as a string.
+ *
+ * @code{.c}
+ * char* txt;
+ * tiledb_array_schema_dump_str(ctx, array_schema, &txt);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param array_schema The array schema.
+ * @param out Returns the output string. Maybe be passed in as nullptr.
+ * @return `TILEDB_OK` for success and `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_array_schema_dump_str(
+    tiledb_ctx_t* ctx, const tiledb_array_schema_t* array_schema, char** out);
```

Comment:
This API has a security defect. There's no maximum length specified.

Personally, I'd rather not have our C API writing directly in user-allocated buffers. That pattern has created lots of systemic mischief (as with user-allocated query buffers). Better would be to have an API for strings with an alloc/free model and have the library manage the memory. We don't have such library-allocated strings ready today, however. (We're needing one already for better error reporting and for delivering warnings, for example.)

We will shortly, with #2748, have experimental features available. This API call could be put into `experimental/` and used immediately pending a library-allocated string.

Reply:
Thanks @eric-hughes-tiledb -- I should add a comment, it does not write into user-allocated buffers. It allocates. (I was, perhaps naively, following the patter of `tiledb_stats_raw_dump`.)

Reply:
How is the deallocation handled?

Regardless of how it's handled, if there's an allocation here, the documentation for this function needs to (1) state that, and (2) say how to deallocate.